### PR TITLE
Features/add hw2

### PR DESCRIPTION
**Pull request by HidirovMarat**

### CODE REVIEW

File: StudenSchool_DO/HW2/FibonacciNumbers.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+﻿namespace HW2
+{
+    internal class FibonacciNumbers : IInformation
+    {
+        private const string Information = "Вывод числа Фибоначчи";
+        private const string MessagesOperation = "Введите порядковый номер числа Фибоначчи:";
+
+        public FibonacciNumbers()
```
Comment (TIove) [inline StudenSchool_DO/HW2/FibonacciNumbers.cs]: Not needed

File: StudenSchool_DO/HW2/DesignMenu.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿namespace HW2
+{
+    internal static class DesignMenu
+    {
+        private static ConsoleColor s_textMenuColor = ConsoleColor.Yellow;
+        private static ConsoleColor s_errorMessagesColor = ConsoleColor.Red;
+        private static ConsoleColor s_serviceMessagesColor = ConsoleColor.Blue;
+        private static ConsoleColor s_everythingElseColor = ConsoleColor.White;
```
Comment (TIove) [inline StudenSchool_DO/HW2/DesignMenu.cs]: `private const ConsoleColor DefaultConsoleColor = ...` is better

File: StudenSchool_DO/HW2/FibonacciNumbers.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
+﻿namespace HW2
+{
+    internal class FibonacciNumbers : IInformation
+    {
+        private const string Information = "Вывод числа Фибоначчи";
```
Comment (TIove) [inline StudenSchool_DO/HW2/FibonacciNumbers.cs]: Information is bad name, rename is likewise 'InformationMessage'

File: StudenSchool_DO/HW2/FibonacciNumbers.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        public FibonacciNumbers()
+        {
+        }
+
+        public string GetInformation()
+        {
+            return Information;
+        }
+
+        public uint GetFibonacciNumbers(uint ordinalNumberOfNumber)
```
Comment (TIove) [inline StudenSchool_DO/HW2/FibonacciNumbers.cs]: Number of number? J

File: StudenSchool_DO/HW2/FibonacciNumbers.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            uint firstFibonacci = 1;
+            uint secondFibonacci = 1;
+            if (ordinalNumberOfNumber == 1)
```
Comment (TIove) [inline StudenSchool_DO/HW2/FibonacciNumbers.cs]: blank line

Comment (TIove) [reply StudenSchool_DO/HW2/FibonacciNumbers.cs]: add breakets {} everywhere

File: StudenSchool_DO/HW2/FibonacciNumbers.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                (firstFibonacci, secondFibonacci) = (secondFibonacci, resultFibonacci);
+            }
+
+            return resultFibonacci;
+        }
+
+        public void Operation()
+        {
+            DesignMenu.WriteTextMenu(MessagesOperation);
+            uint ordinalNumberOfNumber = uint.Parse(Console.ReadLine());
```
Comment (TIove) [inline StudenSchool_DO/HW2/FibonacciNumbers.cs]: blank lines between code

File: StudenSchool_DO/HW2/DesignMenu.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
+﻿namespace HW2
+{
+    internal static class DesignMenu
```
Comment (TIove) [inline StudenSchool_DO/HW2/DesignMenu.cs]: DesignedMenu is better

File: StudenSchool_DO/HW2/FibonacciNumbers.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+            uint resultFibonacci = 0;
+            for (uint i = 3; i < ordinalNumberOfNumber + 1; i++)
+            {
+                try
+                {
+                    resultFibonacci = checked(firstFibonacci + secondFibonacci);
+                }
+                catch(OverflowException overEx)
+                {
+                    throw overEx;
```
Comment (TIove) [inline StudenSchool_DO/HW2/FibonacciNumbers.cs]: You will lose the stack trace here, better to use `throw`

File: StudenSchool_DO/HW2/FileToRead.cs [hunk 9]
```diff
@@ -0,0 +1,52 @@
+﻿using System.Runtime.InteropServices;
+
+namespace HW2
+{
+    internal class FileToRead : IInformation
```
Comment (TIove) [inline StudenSchool_DO/HW2/FileToRead.cs]: FileReader is better

File: StudenSchool_DO/HW2/FileToRead.cs [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            string path = PathOther;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                path = PathWindows;
+
+            if (!File.Exists(path) || specifiedNumber == 0)
+                return new string[] {""};
+
+            string[] linesOfText = File.ReadAllLines(path);
+            return linesOfText[0..specifiedNumber];
```
Comment (TIove) [inline StudenSchool_DO/HW2/FileToRead.cs]: Empty line

File: StudenSchool_DO/HW2/FileToRead.cs [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        }
+
+        public void Operation()
+        {
+            DesignMenu.WriteTextMenu(MessagesOperation);
+            int specifiedNumber = GetSpecifiedNumber();
+            string[] specifiedNumberOfLinesOfText = GetSpecifiedNumberOfLinesOfTextFromFile(specifiedNumber);
+            foreach (var line in specifiedNumberOfLinesOfText)
```
Comment (TIove) [inline StudenSchool_DO/HW2/FileToRead.cs]: Empty line

File: StudenSchool_DO/HW2/FileToRead.cs [hunk 12]
```diff
@@ -0,0 +1,52 @@
+﻿using System.Runtime.InteropServices;
+
+namespace HW2
+{
+    internal class FileToRead : IInformation
+    {
+        private const string Information = "Чтение";
+        private const string MessagesOperation = "Введите сколько строк хотите получить: ";
+        private const string PathWindows = @"..\..\..\Data\FileToRead.txt";
```
Comment (TIove) [inline StudenSchool_DO/HW2/FileToRead.cs]: Get these params from the method's arguments

File: StudenSchool_DO/HW2/FileToRead.cs [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+            return Information;
+        }
+
+        private string[] GetSpecifiedNumberOfLinesOfTextFromFile(int specifiedNumber)
+        {
+            string path = PathOther;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                path = PathWindows;
+
+            if (!File.Exists(path) || specifiedNumber == 0)
```
Comment (TIove) [inline StudenSchool_DO/HW2/FileToRead.cs]: It's better to write some error message, that file is not found

File: StudenSchool_DO/HW2/FileToRead.cs (hunk 10 above)
Comment (TIove) [inline StudenSchool_DO/HW2/FileToRead.cs]: Hmm, you read all file and return only needed lines. Optional, if can, read only needed lines.

File: StudenSchool_DO/HW2/IInformation.cs [hunk 14]
```diff
@@ -0,0 +1,9 @@
+﻿namespace HW2
+{
+    internal interface IInformation
+    {
+        string GetInformation();
+        void Operation();
+    }
+
```
Comment (TIove) [inline StudenSchool_DO/HW2/IInformation.cs:6]: Extra blank line

File: StudenSchool_DO/HW2/Menu.cs [hunk 15]
```diff
@@ -0,0 +1,80 @@
+﻿namespace HW2
+{
+    internal class Menu
+    {
+        private const string UnderMenuMessages = "1. Остаться (выполнить задачу снова) \n2. Вернуться в главное меню";
```
Comment (TIove) [inline StudenSchool_DO/HW2/Menu.cs]: Bad name,  better is OptionalRerunMessage

File: StudenSchool_DO/HW2/WebsiteSave.cs [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+                {
+                    isGood = false;
+                    Console.WriteLine("Введите еще раз url");
+                }
+                if (isGood)
+                    break;
+            }
+        }
+
+
```
Comment (TIove) [inline StudenSchool_DO/HW2/WebsiteSave.cs]: Extra lines

File: StudenSchool_DO/HW2/WebsiteSave.cs [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+                path = PathWindows;
+
+            if (!File.Exists(path))
+                throw new Exception();
+
+            while (true)
+            {
+                bool isGood = true;
+                _url = Console.ReadLine();
+                using WebClient client = new();
```
Comment (TIove) [inline StudenSchool_DO/HW2/WebsiteSave.cs:36]: WebClient is obsolete, use HttpClient instead

File: StudenSchool_DO/HW2/WebsiteSave.cs [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+                try
+                {
+                    client.DownloadFile(_url, path);
+                }
+                catch (Exception ex)
+                {
+                    isGood = false;
```
Comment (TIove) [inline StudenSchool_DO/HW2/WebsiteSave.cs]: use `continue;`

File: StudenSchool_DO/HW2/FileToRead.cs (hunk 11 above)
Comment (HidirovMarat) [reply StudenSchool_DO/HW2/FileToRead.cs]: > Empty line

Why should there be an empty string here?

File: StudenSchool_DO/HW2/FibonacciNumbers.cs (hunk 6 above)
Comment (HidirovMarat) [reply StudenSchool_DO/HW2/FibonacciNumbers.cs]: > blank lines between code
Why should there be an empty lines here?

File: StudenSchool_DO/HW2/WorkWithUser.cs [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+
+        while (!IsValid(out numberOfLines, IsCondition))
+        {
+            DesignedMenu.WriteServiceMessages("Неправильный ввод. Повторите");
+        }
+
+        return numberOfLines;
+    }
+
+    private static bool IsValid(out int value, CheckingCondition IsCondition)
```
Comment (TIove) [inline StudenSchool_DO/HW2/WorkWithUser.cs]: `out` params usually are the last arguments in order. e.g. private static bool IsValid(CheckingCondition IsCondition, out int value)

File: StudenSchool_DO/HW2/Menu.cs [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+            DesignedMenu.WriteTextMenu($"{i + 1}. {_information[i].GetInformation()}");
+        }
+    }
+
+    private void SelectingMenuItem(int numberItem)
+    {
+        do
+        {
+            _information[numberItem].Operate();
+        } while (Repeat());
```
Comment (TIove) [inline StudenSchool_DO/HW2/Menu.cs]: `while` should be on the next line.

**Review by TIove — APPROVED** [review]

File: StudenSchool_DO/HW2/FibonacciNumbers.cs [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+
+        if (ordinalNumber == 1)
+        {
+            return firstFibonacci;
+        }
+
+        if (ordinalNumber == 2)
+        {
+            return secondFibonacci;
+        }
```
Comment (SolovyovDmitry) [inline StudenSchool_DO/HW2/FibonacciNumbers.cs]: can be combined

**Review by SolovyovDmitry — APPROVED** [review]

**Pull request merged**